cps/kni.c

```diff
@@ -0,0 +1,220 @@
+/*
+ * Gatekeeper - DoS protection system.
+ * Copyright (C) 2016 Digirati LTDA.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <fcntl.h>
+#include <unistd.h>
+
+#include <rte_ethdev.h>
+#include <rte_kni.h>
+#include <rte_malloc.h>
+
+#include "gatekeeper_cps.h"
+#include "gatekeeper_main.h"
+#include "kni.h"
+
+#define INSMOD_CMD "insmod "
+#define KNI_KMOD "rte_kni"
+#define RMMOD_CMD ("rmmod " KNI_KMOD)
```

Comment:
1. `INSMOD_CMD` and `RMMOD_CMD` is not being used. Should we drop them?

2. `KNI_KMOD` is only used once in the code. Do we need the define?